src/StatusBar.ts

```diff
@@ -41,32 +77,156 @@ export class StatusBar {
+        if (queueItem) {
+          this.render(queueItem, StatusType.summary)
+          return
+        }
       }
     }
+    this.hideStatusBarItem(StatusType.summary)
   }
-  private render(queueItem: QueueItem) {
-    clearInterval(this.statusBarSpinner)
+  private hideStatusBarItem(statusType: StatusType) {
+    clearInterval(this.getSpinner(statusType))
+    switch (statusType) {
+      case StatusType.active:
+        this.activeStatusItem.hide()
+        break
+      case StatusType.summary:
+        this.summaryStatusItem.hide()
+        break
+      default:
+        throw new Error(`unexpected statusType: ${statusType}`)
+    }
+  }
+  private render(queueItem: QueueItem, statusType: StatusType) {
+    clearInterval(this.getSpinner(statusType))
+
     const message = this.getMessageByStatus(queueItem.status)
     if (queueItem.status === 'running') {
-      this.statusBarSpinner = setInterval(() => this.render(queueItem), 100)
+      this.setSpinner(statusType, () => this.render(queueItem, statusType))
+    }
+
+    switch (statusType) {
+      case StatusType.active:
+        const details = !this.isMultiroot() && queueItem.details ? queueItem.details : ''
+        this.activeStatusItem.text = `Jest: ${message} ${details}`
+        this.activeStatusItem.tooltip = `Jest status of '${this.activeFolder}'`
+        this.activeStatusItem.show()
+        break
+      case StatusType.summary:
+        this.summaryStatusItem.text = `Jest-WS: ${message}`
+        this.summaryStatusItem.show()
+        break
+      default:
+        throw new Error(`unexpected statusType: ${statusType}`)
+    }
+  }
+
+  private getSpinner(type: StatusType) {
+    switch (type) {
+      case StatusType.active:
+        return this.statusBarSpinner.active
+      case StatusType.summary:
+        return this.statusBarSpinner.summary
+      default:
+        throw new Error(`unexpected statusType: ${type}`)
+    }
+  }
+  private setSpinner(type: StatusType, callback: () => void) {
+    const timer = setInterval(callback, 100)
+    switch (type) {
+      case StatusType.active:
+        this.statusBarSpinner.active = timer
+        break
+      case StatusType.summary:
+        this.statusBarSpinner.summary = timer
+        break
+      default:
+        throw new Error(`unexpected statusType: ${type}`)
+    }
+  }
+
+  private updateWorkspaceOutput() {
+    if (!this.workspaceOutput) {
+      this.workspaceOutput = vscode.window.createOutputChannel('Jest (Workspace)')
     }
-    this.statusBarItem.text = `${this.statusKey} ${message} ${queueItem.details || ''}`
+    this.workspaceOutput.clear()
+
+    const messages = this.queue.map(item => {
+      const details = item.details ? `: ${item.details}` : ''
+      return `${item.source}: ${item.status} ${details}`
+    })
+    this.workspaceOutput.append(messages.join('\n'))
   }
+
+  private isMultiroot() {
+    return this.queue.length > 1

```

Comment:
this is implementation detail, it needs to be more explicit

Reply:
not sure what you mean here... I wanted to check if there are multiple sources to decide if to show the summary status. It might make more sense after switching queue to Map (see below)